pytorch_lightning/core/lightning.py

```diff
@@ -31,7 +31,18 @@
 else:
     XLA_AVAILABLE = True
 
-CHECKPOINT_KEY_MODULE_ARGS = 'module_arguments'
+PRIMITIVE_TYPES = (bool, int, float, str)
+ALLOWED_CONFIG_TYPES = (AttributeDict, dict, Namespace)
+try:
+    from omegaconf import DictConfig, OmegaConf
+except ImportError:
+    pass
+else:
+    ALLOWED_CONFIG_TYPES = ALLOWED_CONFIG_TYPES + (DictConfig, OmegaConf)
```

Comment:
OmegaConf is not the config object type.
It is a static class that is used to construct the config objects.

Config objects are be of type:
`omegaconf.Container`
or one of the subclasses:
`omegaconf.{DictConfig, ListConfig}`.

If you are always expecting a dictionary style config use DictConfig (I think this is the case here).